crates/oxc_linter/src/rules/jsx_a11y/img_redundant_alt.rs

```diff
@@ -0,0 +1,245 @@
+
+                    if is_redundant_alt_text(alt_text, &self.redundant_words) {
+                        ctx.diagnostic(ImgRedundantAltDiagnostic(jsx_el.span));
+                    }
+                }
+                Expression::TemplateLiteral(lit) => {
+                    for quasi in lit.quasis.iter() {
+                        let alt_text = quasi.value.raw.as_str();
+
+                        if is_redundant_alt_text(alt_text, &self.redundant_words) {
+                            ctx.diagnostic(ImgRedundantAltDiagnostic(jsx_el.span));
+                        }
+                    }
+                }
+                _ => {
+                    return;
+                }
+            },
+            _ => {
+                return;
+            }
+        };
+    }
+}
+
+fn is_redundant_alt_text(alt_text: &str, redundant_words: &[String]) -> bool {
+    let regexp = Regex::new(&format!(r"(?i)\b({})\b", redundant_words.join("|"),)).unwrap();
+
+    regexp.is_match(alt_text)
+}
+
+#[test]
+fn test() {
+    use crate::tester::Tester;
+
+    fn array() -> serde_json::Value {
+        serde_json::json!([{
+            "components": ["Image"],
+            "words": ["Word1", "Word2"]
+        }])
+    }
+
+    let pass = vec![
+        (r"<img alt='foo' />;", None),
+        (r"<img alt='picture of me taking a photo of an image' aria-hidden />", None),
+        (r"<img aria-hidden alt='photo of image' />", None),
+        (r"<img ALt='foo' />;", None),
+        (r"<img {...this.props} alt='foo' />", None),
+        (r"<img {...this.props} alt={'foo'} />", None),
+        (r"<img {...this.props} alt={alt} />", None),
+        (r"<a />", None),
+        (r"<img />", None),
+        (r"<IMG />", None),
+        (r"<img alt={undefined} />", None),
+        (r"<img alt={`this should pass for ${now}`} />", None),
+        (r"<img alt={`this should pass for ${photo}`} />", None),
+        (r"<img alt={`this should pass for ${image}`} />", None),
+        (r"<img alt={`this should pass for ${picture}`} />", None),
+        (r"<img alt={`${photo}`} />", None),
+        (r"<img alt={`${image}`} />", None),
+        (r"<img alt={`${picture}`} />", None),
+        (r"<img alt={'undefined'} />", None),
+        (r"<img alt={() => {}} />", None),
+        (r"<img alt={function(e){}} />", None),
+        (r"<img aria-hidden={false} alt='Doing cool things.' />", None),
+        (r"<UX.Layout>test</UX.Layout>", None),
+        (r"<img alt />", None),
+        (r"<img alt={imageAlt} />", None),
+        (r"<img alt={imageAlt.name} />", None),
+        (r"<img alt={imageAlt?.name} />", None),
+        (r"<img alt='Doing cool things' aria-hidden={foo?.bar}/>", None),
+        (r"<img alt='Photography' />;", None),
+        (r"<img alt='ImageMagick' />;", None),
+        (r"<Image alt='Photo of a friend' />", None),
+        // TODO we need components_settings to test this
+        // (r"<Image alt='Foo' />", settings: Some(components_settings))
```

Comment:
this is same to [here](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/1adec3517fc2c9797212ca4d38858deed917e7be/__tests__/src/rules/img-redundant-alt-test.js#L26-L32).

I could not find any way to use componentsSettings in linter testing, but I will really appreciate if the way actually exists and you teach me